### PR TITLE
Add option to load pretrained model in TTS

### PR DESCRIPTION
**Pull request by kan-bayashi**
This PR adds options to load pretrained model in TTS.

Related to #1591

### CODE REVIEW

**Comment by kan-bayashi** [comment]
@b-flo I made a little bit change to existing functions.
Could you check it?

File: espnet/bin/tts_train.py [hunk 1]
```diff
@@ -122,6 +122,17 @@ def get_parser():
...
                         help='Whether to keep all data on memory')
+    # finetuning related
+    parser.add_argument('--enc-init', default=None, type=str,
+                        help='Pre-trained TTS model path to initialize encoder.')
+    parser.add_argument('--enc-init-mods', default='enc.',
+                        type=lambda s: [str(mod) for mod in s.split(',') if s != ''],
+                        help='List of encoder modules to initialize, separated by a comma.')
+    parser.add_argument('--dec-init', default=None, type=str,
+                        help='Pre-trained TTS model path to initialize decoder.')
+    parser.add_argument('--dec-init-mods', default='dec.',
```
Comment (b-flo) [inline espnet/bin/tts_train.py]:
```suggestion
    parser.add_argument('--dec-init-mods', default='dec.,att.',
```

Sorry If I'm wrong but don't you use attention in TTS?

File: espnet/asr/pytorch_backend/asr_init.py [hunk 2]
```diff
@@ -147,14 +150,14 @@ def get_trained_model_state_dict(model_path):
 
     Return:
         model.state_dict() (OrderedDict): the loaded model state_dict
-        (str): Type of model. Either ASR/MT or LM.
+        (bool): Flag representing the model is LM or not
```
Comment (b-flo) [inline espnet/asr/pytorch_backend/asr_init.py]:
```suggestion
        (bool): Boolean defining whether the model is an LM
```

File: espnet/bin/tts_train.py (hunk 1 above)
Comment (kan-bayashi) [reply espnet/bin/tts_train.py]: `att` is included in `dec` (`dec.att`).
Therefore, `dec` is enough for TTS.

File: espnet/asr/pytorch_backend/asr_init.py [hunk 3]
```diff
@@ -35,9 +36,11 @@ def transfer_verification(model_state_dict, partial_state_dict, modules):
     return len(partial_modules) > 0
 
 
-def get_partial_asr_mt_state_dict(model_state_dict, modules):
+def get_partial_state_dict(model_state_dict, modules):
     """Create state_dict with specified modules matching input model modules.
 
+    Note that please use get_partial_lm_state_dict for LM.
```
[the remark this answers is not in the document]
Comment (kan-bayashi) [reply espnet/asr/pytorch_backend/asr_init.py]: Hmm. This is a kind of reminder comment for us.

File: espnet/bin/tts_train.py (hunk 1 above)
Comment (b-flo) [reply espnet/bin/tts_train.py]: Got it, thanks! Resolving the conversation then!

**Comment by kan-bayashi** [comment]
Thanks, @b-flo. I will merge.

**Pull request merged**